micro-rdk/src/esp32/conn/network.rs

```diff
@@ -202,25 +202,30 @@ impl Esp32WifiNetwork {
 
         let sl_stack = esp32_get_system_event_loop()?;
 
-        let subscription = sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| {
-            if matches!(event, WifiEvent::StaDisconnected(_)) {
-                if let Ok(wifi) = esp32_get_wifi() {
-                    if let Some(mut wifi_guard) = wifi.try_lock() {
-                        let wifi_mut = wifi_guard.wifi_mut();
-                        if let Err(err) = wifi_mut.connect() {
-                            let ssid = wifi_mut
-                                .get_configuration()
-                                .map_or("<no_ssid>".to_owned(), |c| {
-                                    c.as_client_conf_ref().unwrap().ssid.to_string()
-                                });
-                            log::error!("could not connect to WiFi {} cause : {:?}", ssid, err);
+        let subscription =
+            sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| match event {
+                WifiEvent::StaDisconnected(disconnected) => {
+                    let reason: super::wifi_error::WifiErrReason = disconnected.reason().into();
+                    log::info!("StaDisconnected event received: {:?}", reason);
```

Comment:
I wouldn't bother writing `StaDisconnected event` here - it isn't really a meaningful bit of information for a consumer of the logs.

Instead, let's focus the log on user facing info: let's log the SSID, the reason, and, the RSSI. Something like:

```
log::info!("received a WiFi disconnection event for SSID {} (RSSI {}) with reason {}"
```